### PR TITLE
Food-Recipe-App: Completion

### PR DESCRIPTION
**Pull request by Megagig**
**In this Milestone, we implemented the following project Requirements:**

**Project requirements**

- You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.

- Login page and registration page:

- Should be built with Devise.
- Recipes list:

- Should display a list of recipes created by the logged-in user as in the wireframe.
- Should lead to recipe details.
- Public recipe list:

- Should display a list of all public recipes ordered by newest as in the wireframe.
- Should lead to recipe details.
- If the user is the owner of the recipe, should allow the user to delete it.
- Recipe details:

- Should display a toggle button that allows for a recipe to be made public or private.
- If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
- If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- Make sure there are no N+1 queries happening.

- Create a navigation menu that allows users to open all of the pages you created.

- Write unit and integration tests

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,18 @@
...
+  # protected
+
+  # def update_allowed_parameters
+  #   devise_parameter_sanitizer.permit(:sign_up) do |u|
+  #     u.permit(:name, :email, :password, :password_confirmation)
+  #   end
+  #   devise_parameter_sanitizer.permit(:account_update) do |u|
+  #     u.permit(:name, :email, :password, :password_confirmation, :current_password)
+  #   end
+  # end
```
Comment (mckent05) [inline app/controllers/application_controller.rb:17]: - You have done well so far, but please restore back this piece of code, having configured paramaters/strong parameters help to ensure that the database is protected against malicious attacks. Please fix this 👍

File: app/controllers/general_shoping_controller.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    @total_amount = 0
+    @count = 0
+    @food_items.each do |food_item|
+      food_item.quantity = food_item.quantity.negative? ? food_item.quantity.abs : food_item.quantity
+      amount = food_item.quantity * food_item.price
+      @total_amount += amount
+      @count += 1
+    end
+  end
+end
```
Comment (mckent05) [inline app/controllers/general_shoping_controller.rb:14]: - You have done well so far, but please fix the logic in your general shopping controller, it is not ideal for users to iadd a food with negative quantity. Please implement this according to the project requirements.
eththstpin

- 
- 
-

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Megagig, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job working on your rails group project 👍 👍 


## Required Changes ♻️
**- Please add more integration tests for your food page, general_shopping list page, this would ensure that your application is properly tested**
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/general_shoping_controller.rb (hunk 2 above)
Comment (jubaer919) [reply app/controllers/general_shoping_controller.rb:14]: @mckent05 my controller already have quantity more than 0(line 4).....  fix the issues  with RecipeFood  validation....

**Review by Efiamotu-1 — APPROVED** [review]
Hi @Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# STATUS: APPROVED 🕺🥳 

image

## Highlights :dart: 
* Correct Gitflow :heavy_check_mark: 
* Good commit messages :heavy_check_mark: 
* All linters passing :heavy_check_mark: 
* Good PR summary :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, you can now Merge since it is Approved ✅, However if there are any conflicts please resolve them before merging.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**